software/deploy/deploy_test.go

```diff
@@ -113,15 +113,29 @@ func (s *Suite) SetupSuite() {
 }
 
 func (s *Suite) SetupTest() {
+	testUtil.InitTestConfig(testUtil.SoftwarePlatform)
+	s.mockImageHandler = new(mocks.ImageHandler)
+	imageHandlerInit = func(image string) airflow.ImageHandler {
+		return s.mockImageHandler
+	}
+	s.houstonMock = new(houston_mocks.ClientInterface)
+}
+
+func (s *Suite) SetupSubTest() {
+	testUtil.InitTestConfig(testUtil.SoftwarePlatform)
 	s.mockImageHandler = new(mocks.ImageHandler)
 	imageHandlerInit = func(image string) airflow.ImageHandler {
 		return s.mockImageHandler
 	}
 	s.houstonMock = new(houston_mocks.ClientInterface)
 }
 
+func (s *Suite) TearDownSubTest() {
+	s.houstonMock.AssertExpectations(s.T())
```

Comment:
nit: Thanks for incorporating, we could now drop the `s.houstonMock.AssertExpectations(s.T())` and `s.mockImageHandler.AssertExpectations(s.T())` from other tests in this suite as well. Can we clean up the code 🙂

Reply:
I can create a seperate ticket and PR for that but I don't want to change the other unrelated functions in this PR itself.

Reply:
I won't say it's unrelated. Also, it's enclosed to unit tests and there is no functional change. But if you want to do it in a separate issue and PR, I would expect that issue linked here and PR to come up sooner rather than going into a backlog.

Reply:
It's unrelated because the goal of this PR is to introduce --remote flag and not to update existing tests (unless those tests are somehow related to the functionality change in this PR).
I've created this - https://github.com/astronomer/issues/issues/6862